fix: track spans in member expression properties for accurate sourcemaps

IWANABETHATGUY opened the pull request:
Closed #6099

The issue was already solved by #6091. This pr adds a test case for the sourcemapping generation

IWANABETHATGUY commented:
* **#6100** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6100?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6100?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#6091** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6091?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite: merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/rolldown?ref=merge-queue-instructions-in-stack-comment&prId=6100).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

sapphi-red commented on crates/rolldown/tests/rolldown/sourcemap/issue_6099/artifacts.snap:
```diff
@@ -0,0 +1,41 @@
+(0:18) "\"1.0.0\"\n" --> (28:19) "\"1.0.0\";\n"
+- ../main.js
+(2:0) "const " --> (34:0) "const "
+(2:6) "foo = react.version.startsWith(" --> (34:6) "foo = import_react.version.startsWith("
```
I think this should be:
```
"foo = react." --> "foo = import_react."
"version." --> "version."
"startsWith(" --> "startsWith("
```
otherwise the stacktrace becomes different.

[REPL](https://repl.rolldown.rs/#eNptUDFuwzAM/AqhRQ5gqOlqoGPnduuixXXoRKkiGhQdpzD899J2HGToIAgn3unuOJrWVKNhivFAQ3INpTYc3TnPr8lU/01K0+gAbx2xwAHbuo8Co08A1EvXS7UCgEw9N3ipuwpsSDEktPNg8mkyU2lCOuDtyeqBV4NwWQwY60agZbqAdS8LUo717NN8NFbWMRG8rVR3Rc6BkstSs+SvIKfCm1dvdhudIrpIx0JFy5v6oamEe9RQm8Gj/4afW+fNRE31b7d3e2+2RB/fZ1SJLkYLfzJ1yPJb3IUl2LvUlrompQMcUYrdBgDkxDRAwgHemYkLmxFByzQ/wnWDdsk8b1GvaS0wafCrpou1YBYz/QGm8aCD)
For this example, `node input.js` outputs
```
Error: see stacktrace
    at Object.get [as version] (C:\Users\green\Downloads\foo\react.js:5:11)
    at file:///C:/Users/green/Downloads/foo/input.js:3:19
```
while `node --enable-source-maps output.js` outputs
```
Error: see stacktrace
    at Object.get (C:\Users\green\Downloads\react.js:5:11)
    at C:\Users\green\Downloads\foo\output.js:15:21
    at Object.<anonymous> (C:\Users\green\Downloads\index.js:3:13)
```
(`3:13` should be `3:19`).

sapphi-red commented:
Maybe we can improve the test infra to verify the stacktrace is same like https://github.com/oxc-project/oxc/pull/13351 does.

IWANABETHATGUY committed:
fix: track spans in member expression properties for accurate sourcemaps (#6100)

Closed #6099

The issue was already solved by #6091. This pr adds a test case for the sourcemapping generation
